driver-sync/src/test/functional/com/mongodb/client/AbstractClientSideEncryptionKmsTlsTest.java

```diff
@@ -116,5 +118,103 @@ public void testInvalidKmsCertificate() {
             assertTrue(expectedKmsTlsError.causeContainsExpectedMessage(e));
         }
     }
+
+    @Test()
+    public void testThatCustomSslContextIsUsed() {
+        assumeTrue(serverVersionAtLeast(4, 2));
+        assumeTrue(hasEncryptionTestsEnabled());
+
+        Map<String, Map<String, Object>> kmsProviders = getKmsProviders();
+        ClientEncryptionSettings clientEncryptionSettings = ClientEncryptionSettings.builder()
+                .keyVaultMongoClientSettings(getMongoClientSettings())
+                .keyVaultNamespace("keyvault.datakeys")
+                .kmsProviders(kmsProviders)
+                .kmsProviderSslContextMap(new HashMap<String, SSLContext>() {{
+                    put("aws", getUntrustingSslContext());
+                    put("azure", getUntrustingSslContext());
+                    put("gcp", getUntrustingSslContext());
+                    put("kmip", getUntrustingSslContext());
+                }})
+                .build();
+        try (ClientEncryption clientEncryption = getClientEncryption(clientEncryptionSettings)) {
+            for (String curProvider: kmsProviders.keySet()) {
+                MongoClientException e = assertThrows(MongoClientException.class, () ->
+                        clientEncryption.createDataKey(curProvider, new DataKeyOptions().masterKey(
+                                BsonDocument.parse(getMasterKey(curProvider)))));
+                assertTrue(e.getMessage().contains("Don't trust anything"));
```

Comment:
Matching against the message in the exception thrown from line 208, proving that the SSLContext is being used